File: 0048-rotate-image/0048-rotate-image.scala
object Solution {
    def rotate(matrix: Array[Array[Int]]): Unit = {

    // transpose

        var n = matrix.length

        for(i<-0 until n){
            for(j<-0 until n){
                if(i>=j){
                val temp = matrix(i)(j)
                matrix(i)(j) = matrix(j)(i)
                matrix(j)(i) =  temp
            }
        }
        }

        for(i<-0 until n){

            var lp = 0
            var rp = n-1
            while(lp<=rp){
                val temp = matrix(i)(lp)
                matrix(i)(lp) = matrix(i)(rp)
                matrix(i)(rp) = temp
                    lp+=1
                    rp-=1
                }
        }



    }
}

File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        
        n = len(matrix)

        for i in range(n):
            for j in range(n):

                if i>=j:
                    matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]

        for i in range(n):

            lp, rp = 0, n-1

            while lp<=rp:
                matrix[i][lp], matrix[i][rp] = matrix[i][rp], matrix[i][lp]
                lp+=1
                rp-=1
